lib/blocks.php

```diff
@@ -93,3 +93,85 @@ function do_blocks( $content ) {
 	return $content_after_blocks;
 }
 add_filter( 'the_content', 'do_blocks', 9 ); // BEFORE do_shortcode() and wpautop().
+
+/**
+ * Extract the blocks from post content for the REST API post response.
+ *
+ * @since 1.1.0
+ *
+ * @param string The post content.
+ *
+ * @return array Array of block data.
+ */
+function get_block_data_for_api_from_post_content( $content ) {
+	$registry = WP_Block_Type_Registry::get_instance();
+	$blocks   = gutenberg_parse_blocks( $content );
+	$data     = array();
+
+	// Loop thru the blocks, adding rendered content when available.
+	foreach ( $blocks as $block ) {
+		$block_name  = isset( $block['blockName'] ) ? $block['blockName'] : null;
+		$attributes  = is_array( $block['attrs'] ) ? $block['attrs'] : null;
+		$raw_content = isset( $block['rawContent'] ) ? $block['rawContent'] : null;
+		if ( null !== $block_name ) {
+			$block_type = $registry->get_registered( $block_name );
+			if ( null !== $block_type ) {
+				$block['renderedContent'] = $block_type->render( $attributes, $raw_content );
+			}
+
+			// Set up the item data.
+			$item_data = array();
+			$item_data['type'] = $block['blockName'];
+			if ( null !== $attributes ) {
+				$item_data['attributes'] = $block['attrs'];
+			}
+			$item_data['content'] = $block['rawContent'];
+			if ( null !== $block['renderedContent'] ) {
+				$item_data['rendered'] = $block['renderedContent'] ;
+			}
+
+			$data[] = $item_data;
+		}
+	}
+
+	return $data;
+}
+
+/**
+ * Attach a post's block data callback to the REST API response.
+ *
+ * @since 1.1.0
+ *
+ * @param string | array $post_type Post type, or array of post types.
+ */
+function attach_block_response_callback( $post_type ) {
+	if ( empty( $post_type ) ) {
+		$post_type = 'post';
+	}
+	if ( is_array( $post_type ) ) {
+		foreach ( $post_type as $type ) {
+			add_filter( 'rest_prepare_' . $type, 'attach_block_data_to_post_response', 10, 2 );
+		}
+	} else {
+		add_filter( 'rest_prepare_' . $post_type, 'attach_block_data_to_post_response', 10, 2 );
+	}
+}
```

Comment:
Wrapping in an array would be cleaner here:

```
if ( ! is_array( $post_type ) ) {
    $post_type = array( $post_type );
}
```

Reply:
Brilliant, thank you! Fixed in https://github.com/WordPress/gutenberg/pull/2649/commits/8340341c7257a975d8895a3c85f78979d833280c